msm/src/ffa/constraint.rs

```diff
@@ -1,62 +1,61 @@
 use crate::{
     columns::{Column, ColumnIndexer},
     expr::MSMExpr,
-    ffa::columns::FFAColumnIndexer,
-    {Fp, N_LIMBS},
+    ffa::{columns::FFAColumnIndexer, interpreter::FFAInterpreterEnv},
 };
+use ark_ff::PrimeField;
 use kimchi::circuits::{
-    expr::{ConstantExprInner, ExprInner, Operations, Variable},
+    expr::{ConstantExpr, ConstantTerm, Expr, ExprInner, Variable},
     gate::CurrOrNext,
 };
 
-/// Access exprs for addition
-pub fn get_exprs_add() -> Vec<MSMExpr<Fp>> {
-    let mut limb_exprs: Vec<_> = vec![];
-    for i in 0..N_LIMBS {
-        let limb_constraint = {
-            let a_i = MSMExpr::Atom(
-                ExprInner::<Operations<ConstantExprInner<Fp>>, Column>::Cell(Variable {
-                    col: FFAColumnIndexer::A(i).ix_to_column(),
-                    row: CurrOrNext::Curr,
-                }),
-            );
-            let b_i = MSMExpr::Atom(ExprInner::Cell(Variable {
-                col: FFAColumnIndexer::B(i).ix_to_column(),
-                row: CurrOrNext::Curr,
-            }));
-            let c_i = MSMExpr::Atom(ExprInner::Cell(Variable {
-                col: FFAColumnIndexer::C(i).ix_to_column(),
-                row: CurrOrNext::Curr,
-            }));
-            a_i + b_i - c_i
-        };
-        limb_exprs.push(limb_constraint);
-    }
-    limb_exprs
+/// Contains constraints for just one row.
+pub struct ConstraintBuilderEnv<F> {
+    pub constraints: Vec<MSMExpr<F>>,
 }
 
-/// Get expressions for multiplication
-pub fn get_exprs_mul() -> Vec<MSMExpr<Fp>> {
-    let mut limb_exprs: Vec<_> = vec![];
-    for i in 0..N_LIMBS {
-        let limb_constraint = {
-            let a_i = MSMExpr::Atom(
-                ExprInner::<Operations<ConstantExprInner<Fp>>, Column>::Cell(Variable {
-                    col: FFAColumnIndexer::A(i).ix_to_column(),
-                    row: CurrOrNext::Curr,
-                }),
-            );
-            let b_i = MSMExpr::Atom(ExprInner::Cell(Variable {
-                col: FFAColumnIndexer::B(i).ix_to_column(),
-                row: CurrOrNext::Curr,
-            }));
-            let d_i = MSMExpr::Atom(ExprInner::Cell(Variable {
-                col: FFAColumnIndexer::D(i).ix_to_column(),
-                row: CurrOrNext::Curr,
-            }));
-            a_i * b_i - d_i
-        };
-        limb_exprs.push(limb_constraint);
+impl<F: PrimeField> FFAInterpreterEnv<F> for ConstraintBuilderEnv<F> {
+    type Position = Column;
+
+    type Variable = MSMExpr<F>;
```

Comment:
Personal taste: I would simply use `E<ConstrantExpr<F>, Column>` to have something generic.

Reply:
Yeah, maybe it's better. I want to kind of highlight that we're using `Column` that is an MSM type (generic, and not particular), and thus the expression is not a generic expression, but a particular over this generic column. Not sure maybe it's already clear though...

Reply:
`Expr<ConstantExpr<F>,crate::column::Column>` is a good alternative too.